services/motion/builtin/move_request.go

```diff
@@ -864,3 +965,61 @@ func toGeoPosePlanSteps(posesByComponent []motionplan.PlanStep, geoPoses []spati
 	}
 	return steps, nil
 }
+
+// getCurrentPosition returns a kinematic bases current position with respect to the world frame.
+func (mr *moveRequest) getCurrentPosition(ctx context.Context) (*referenceframe.PoseInFrame, error) {
+	// get the current position of the base which we will use to transform the detection into world coordinates
+	currentPosition, err := mr.kinematicBase.CurrentPosition(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// make sure the currentPosition is in the world frame
+	if currentPosition.Parent() != referenceframe.World {
+		// get the current inputs from the frame system service
+		currentInputs, _, err := mr.fsService.CurrentInputs(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		// transform using the plan request frame system
+		tf, err := mr.planRequest.FrameSystem.Transform(currentInputs, currentPosition, referenceframe.World)
+		if err != nil {
+			return nil, err
+		}
+
+		// assert back to PoseInFrame
+		var ok bool
+		if currentPosition, ok = tf.(*referenceframe.PoseInFrame); !ok {
+			return nil, errors.New("cannot assert transformable as PoseInFrame")
+		}
+	}
+	return currentPosition, nil
+}
+
+// getExistingGeometries returns elements of the worlstate which are not transient.
+// Depending on the caller, the geometries returned are in their relative position to the base or
+// in their absolute position with respect to the world.
+func (mr *moveRequest) getExistingGeometries(
+	f func(g spatialmath.Geometry) spatialmath.Geometry,
+) (*referenceframe.GeometriesInFrame, error) {
+	// get the already defined geometries of worldstate
+	existingGifs, err := mr.planRequest.WorldState.ObstaclesInWorldFrame(
+		mr.planRequest.FrameSystem, mr.planRequest.StartConfiguration,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// existing geometries are in their relative position, i.e. with respect to the base's frame
+	// here we transform them into their absolute positions. i.e. with respect to the world frame
+	// Note: we remove any existing transient obstacle geometries as they will be re-detected
+	existingGeoms := []spatialmath.Geometry{}
+	for _, g := range existingGifs.Geometries() {
```

Comment:
> Why not just treat these separately to begin with?

Can you please expand on this? I am not sure what you mean here.

--------------------

We add transient obstacles to the world state for them to considered when planning. 
However, if there were a replan, the positions of the transient obstacles themselves is likely outdated so here we remove them and allow new ones to take their place.

Reply:
I think the issue here is that the worldState is being treated statefully and mutated which it wasnt designed to support.  My expectation for how this would look is that
1.  we have an existing worldState AKA static geometries
2. we get detections from cameras
3. we combine the two into a single worldState object that is used for the duration of a plan 

Then if we have to make a new plan we would do this process again rather than mutating the existing worldState.

Reply:
ah ok, this makes sense to me. Thanks for explaining!